c-gull/Cargo.toml

```diff
@@ -11,19 +11,20 @@ repository = "https://github.com/sunfishcode/mustang"
 edition = "2021"
 
 [dependencies]
-rustix = { version = "0.36.1", default-features = false, features = ["fs", "itoa", "net", "param", "process", "rand", "termios", "thread", "time"] }
 # We use the libc crate for C ABI types and constants, but we don't depend on
 # the actual platform libc.
 libc = { version = "0.2.138", default-features = false }
-errno = { version = "0.2.8", default-features = false }
-c-scape = { path = "../c-scape", version = "0.6.1", features = ["std"] }
-tz-rs = "0.6.11"
-printf-compat = "0.1.1"
-log = { version = "0.4.14", default-features = false }
+c-scape = { path = "../c-scape", version = "0.6.1", default-features = false }
+errno = { version = "0.2.8", default-features = false, optional = true }
+tz-rs = { version = "0.6.11", optional = true }
+printf-compat = { version = "0.1.1", optional = true}
+log = { version = "0.4.14", default-features = false, optional = true }
+rustix = { version = "0.36.1", default-features = false, optional = true }
 
 [dev-dependencies]
 libc = "0.2.138"
 
 [features]
-default = ["threads"]
+default = ["threads", "std"]
 threads = ["c-scape/threads"]
+std = ["c-scape/default", "c-scape/std", "rustix/fs", "rustix/itoa", "rustix/net", "rustix/param", "rustix/process", "rustix/rand", "rustix/termios", "rustix/thread", "rustix/time", "log", "printf-compat", "tz-rs", "errno"]
```

Comment:
In theory, many of these rustix features should work in no-std mode, so they could still be specified in the `[dependencies]` section instead of here in the `[features]` section. That said, I don't know offhand how well that works at the moment.

Reply:
I switched rustix to an optional dependency which makes this much cleaner.